core/modules/ccontrol/UserQueryFactory.cc

```diff
@@ -154,28 +156,49 @@ UserQueryFactory::newUserQuery(std::string const& aQuery,
         std::string errorExtra;
 
         // Parse SELECT
-        std::shared_ptr<query::SelectStmt> a4stmt = a4NewUserQuery(query);
+
+        // parse using antlr4
+        std::shared_ptr<query::SelectStmt> a4stmt;
+        try {
+            a4stmt = a4NewUserQuery(query);
+        } catch (std::exception& e) {
+            LOGS(_log, LOG_LVL_ERROR, "Antlr4 error: " << e.what());
+            return std::make_shared<UserQueryInvalid>(std::string("ParseException:") + e.what());
+        }
         if (a4stmt) {
             LOGS(_log, LOG_LVL_DEBUG, "Antlr4 generated select statement: " << a4stmt->getQueryTemplate());
             LOGS(_log, LOG_LVL_DEBUG, "Antlr4-style Hierarchy: " << *a4stmt);
         } else {
             LOGS(_log, LOG_LVL_DEBUG, "Antlr4 did not generate a select statement.");
         }
 
-        std::shared_ptr<query::SelectStmt> stmt;
+        // parse using antlr v2
+        std::shared_ptr<query::SelectStmt> a2stmt;
         try {
-            stmt = antlr2NewSelectStmt(query);
+            a2stmt = antlr2NewSelectStmt(query);
         } catch(parser::ParseException const& e) {
             return std::make_shared<UserQueryInvalid>(std::string("ParseException:") + e.what());
         }
-        LOGS(_log, LOG_LVL_DEBUG, "Old-style generated select statement: " << stmt->getQueryTemplate());
-        LOGS(_log, LOG_LVL_DEBUG, "Old-style Hierarchy: " << *stmt);
+        LOGS(_log, LOG_LVL_DEBUG, "Old-style generated select statement: " << a2stmt->getQueryTemplate());
+        LOGS(_log, LOG_LVL_DEBUG, "Old-style Hierarchy: " << *a2stmt);
 
         // TEMP while developing the antlr4 parser listener
-        if (a4stmt && stmt) {
-            bool theyMatch(*a4stmt == *stmt);
-            LOGS(_log, LOG_LVL_DEBUG, "they match:" << theyMatch);
+        if (a4stmt && a2stmt) {
+            bool theyMatch(*a4stmt == *a2stmt);
+            LOGS(_log, LOG_LVL_DEBUG, "antlr v2 and antlr4 generated queries match:" << theyMatch);
+        }
+
+        // TEMP while developing the antlr4 query generation; decide which generated select statment to use
+        // to process the query:
+#if USE_ANTLR4_QUERY
```

Comment:
Would it make a sense to avoid making a2stmt if this is defined? The #ifdef could be moved up to line 159.

Reply:
for the time being it makes sense to generate (and log) both statements, because I often compare them to each other when problems arise.